packages/node/src/e2e/e2e-test.spec.ts

```diff
@@ -36,157 +36,201 @@ config({
   silent: process.env.CI === "true",
 });
 
-// This first test just saves the trouble of looking for a library failure when
-// the environment wasn't properly set.
-describe("Environment", () => {
-  it("contains the expected environment variables", () => {
-    expect(process.env.E2E_TEST_CLIENT_ID).not.toBeUndefined();
-    expect(process.env.E2E_TEST_CLIENT_SECRET).not.toBeUndefined();
-    expect(process.env.E2E_TEST_IDP_URL).not.toBeUndefined();
-  });
-});
-
-describe("Authenticated fetch", () => {
-  it("properly sets up session information", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
-    });
-    expect(session.info.isLoggedIn).toEqual(true);
-    expect(session.info.sessionId).not.toBeUndefined();
-    expect(session.info.webId).not.toBeUndefined();
-  });
-
-  it("can fetch a public resource when logged in", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
-    });
-    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    const publicResourceUrl = session.info.webId!;
-    const response = await session.fetch(publicResourceUrl);
-    expect(response.status).toEqual(200);
-    await expect(response.text()).resolves.toContain(
-      ":PersonalProfileDocument"
-    );
-  });
-
-  it("can fetch a private resource when logged in", async () => {
-    const session = new Session();
-    await session.login({
-      clientId: process.env.E2E_TEST_CLIENT_ID,
-      clientSecret: process.env.E2E_TEST_CLIENT_SECRET,
-      oidcIssuer: process.env.E2E_TEST_IDP_URL,
+type OidcIssuer = string;
+type ClientId = string;
+type ClientSecret = string;
+type Pod = string;
+type AuthDetails = [Pod, OidcIssuer, ClientId, ClientSecret];
+// Instructions for obtaining these credentials can be found here:
+// https://github.com/inrupt/solid-client-authn-js/blob/1a97ef79057941d8ac4dc328fff18333eaaeb5d1/packages/node/example/bootstrappedApp/README.md
+const serversUnderTest: AuthDetails[] = [
+  // pod.inrupt.com:
+  [
+    // Cumbersome workaround, but:
+    // Trim `https://` from the start of these URLs,
+    // so that GitHub Actions doesn't replace them with *** in the logs.
+    process.env.E2E_TEST_ESS_POD!.replace(/^https:\/\//, ""),
```

Comment:
Can we extract those replace into a utility function?

```javascript
function trimHttps(x: string): string {
  return x.replace(/^https:\/\//, "");
}
```

Reply:
Probably, but do you think that would make the code more readable ?

Reply:
I think it would and more importantly, I opt for DRY by default; `.replace(/^https:\/\//, "")` is arguably a case of code duplication.

However, it's pretty easy to read as is and the "repeat" part only holds because of the generic regex... So frankly I don't mind. ^^,

(I'm much more bothered by all of our CI being a big case of code duplication (and by our, I don't mean it's endemic))

I'll mark as resolved, we have bigger fishes to fry ;)

Reply:
Yeah, there IS a lot of duplication across our CI, but I'm not sure how it could be done otherwise... Another thing to consider on the long run :)